src/Providers/Models/DTO/ModelConfig.php

```diff
@@ -0,0 +1,721 @@
+    }
+
+    /**
+     * Gets whether to return log probabilities.
+     *
+     * @since n.e.x.t
+     *
+     * @return bool|null Whether to return log probabilities.
+     */
+    public function getLogprobs(): ?bool
+    {
+        return $this->logprobs;
+    }
+
+    /**
+     * Sets the number of top log probabilities to return.
+     *
+     * @since n.e.x.t
+     *
+     * @param int $topLogprobs The number of top log probabilities.
+     * @return self For method chaining.
+     */
+    public function setTopLogprobs(int $topLogprobs): self
+    {
+        $this->topLogprobs = $topLogprobs;
+        return $this;
+    }
+
+    /**
+     * Gets the number of top log probabilities to return.
+     *
+     * @since n.e.x.t
+     *
+     * @return int|null The number of top log probabilities.
+     */
+    public function getTopLogprobs(): ?int
+    {
+        return $this->topLogprobs;
+    }
+
+    /**
+     * Sets the tools.
+     *
+     * @since n.e.x.t
+     *
+     * @param Tool[] $tools The tools.
+     * @return self For method chaining.
+     */
+    public function setTools(array $tools): self
+    {
+        $this->tools = $tools;
+        return $this;
+    }
+
+    /**
+     * Gets the tools.
+     *
+     * @since n.e.x.t
+     *
+     * @return Tool[]|null The tools.
+     */
+    public function getTools(): ?array
+    {
+        return $this->tools;
+    }
+
+    /**
+     * Sets the custom options.
+     *
+     * @since n.e.x.t
+     *
+     * @param array<string, mixed> $customOptions The custom options.
+     * @return self For method chaining.
+     */
+    public function setCustomOptions(array $customOptions): self
+    {
+        $this->customOptions = $customOptions;
+        return $this;
+    }
+
+    /**
+     * Gets the custom options.
+     *
+     * @since n.e.x.t
+     *
+     * @return array<string, mixed>|null The custom options.
+     */
+    public function getCustomOptions(): ?array
+    {
+        return $this->customOptions;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public static function getJsonSchema(): array
+    {
+        return [
+            'type' => 'object',
+            'properties' => [
+                'outputModalities' => [
+                    'type' => 'array',
+                    'items' => [
+                        'type' => 'string',
+                        'enum' => ModalityEnum::getValues(),
+                    ],
+                    'description' => 'Output modalities for the model.',
+                ],
+                'systemInstruction' => [
+                    'type' => 'string',
+                    'description' => 'System instruction for the model.',
+                ],
+                'candidateCount' => [
+                    'type' => 'integer',
+                    'minimum' => 1,
+                    'description' => 'Number of response candidates to generate.',
+                ],
+                'maxTokens' => [
```

Comment:
I'm only realizing this now, but I think we should add class constants to every DTO for each key used in the DTO's array representation.

Having constants makes it easier to work with the array data, no chance for typos and so on.

For `ModelConfig` this is especially important, because those string values will also be relevant for the `RequiredOption` / `SupportedOption` functionality.

But I think we should add those constants to all DTOs.